pkg/plugins/pg/source/cdc.go

```diff
@@ -28,9 +28,24 @@ import (
 	"github.com/jackc/pgx"
 )
 
+var bufferSize = 1000
+
 // withCDC sets up change data capture for the Postgres Source or returns an
 // error.
-func (s *Source) withCDC(ctx context.Context, cfg plugins.Config) error {
+func (s *Source) withCDC(cfg plugins.Config) error {
+	c := context.Background()
+	ctx, cancel := context.WithCancel(c)
+	s.killswitch = cancel
+	// early return if cdc is disabled
+	v, ok := cfg.Settings["cdc"]
+	// if cdc is set, check the value for falsy values.
+	if ok {
+		switch v {
+		case "disabled", "false", "off", "0":
```

Comment:
Do we care about case sensitivity here?

Reply:
We probably do, but I don't know at what level to enforce that at. I would think we should address that in a different PR.

Reply:
Yup, that makes sense. Would be nice if we capture this in an issue, so we don't lose track of it.

Reply:
Hmm, I'd rather use sort of an enum here and allow only two values (e.g. `true`/`false` or `enabled`/`disabled` or `on`/`off`), if the field contains anything else we return an error, the config is invalid, I wouldn't want to silently fall back if the user specified an invalid config option.

We should standardize this by providing some helper methods in the SDK for parsing the config and ensuring we use consistent values and error messages across plugins. No need to act on this now, just writing down my thoughts, I'll create an issue for this when I get the time.